src/test/java/com/muzika/homeworksb/controller/TodoControllerTest.java

```diff
@@ -0,0 +1,276 @@
+        TodoCreateDto requestDto = new TodoCreateDto(
+            "Wake up",
+            "",
+            LocalDateTime.of(2024, 12, 16, 11, 30),
+            PriorityEnum.LOW
+        );
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        Authentication authenticatedUser = SecurityContextHolder.getContext().getAuthentication();
+
+        // Act
+        MvcResult result = mockMvc.perform(post("/todos")
+                .principal(authenticatedUser)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRequest))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        // Assert
+        TodoResponseDto responseDto = objectMapper.readValue(result.getResponse().getContentAsString(), TodoResponseDto.class);
+        Assertions.assertNotNull(responseDto);
+        Assertions.assertNotNull(responseDto.id());
+        Assertions.assertNotNull(responseDto.title());
+        Assertions.assertNotNull(responseDto.description());
+        Assertions.assertNotNull(responseDto.dueDate());
+        Assertions.assertNotNull(responseDto.priority());
+        Assertions.assertNull(responseDto.status());
+        Assertions.assertNotNull(responseDto.createdDate());
+        Assertions.assertNull(responseDto.updatedDate());
+        Assertions.assertNotNull(responseDto.userId());
+    }
+
+    @Test
+    @DisplayName("Create a new todo with not existed priority - failed request")
+    void createTodo_WrongPriority_Fail() throws Exception {
+        // Arrange
+        String invalidRequest = """
+        {
+            "title": "Wake up",
+            "dueDate": "2024-12-16T11:30:00",
+            "priority": "111"
+        }
+        """;
+
+        // Act
+        MvcResult result = mockMvc.perform(post("/todos")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(invalidRequest))
+            .andExpect(status().isBadRequest())
+            .andReturn();
+
+        // Assert
+        String responseContent = result.getResponse().getContentAsString();
+        ObjectMapper objectMapper = new ObjectMapper();
+        ErrorResponse errorResponse = objectMapper.readValue(responseContent, ErrorResponse.class);
+
+        Assertions.assertEquals("Bad Request", errorResponse.title());
+        Assertions.assertEquals(400, errorResponse.status());
+        Assertions.assertEquals("/todos", errorResponse.instance());
+        Assertions.assertEquals("Failed to read request", errorResponse.detail());
+    }
+
+    public record ErrorResponse(
+        String type,
+        String title,
+        int status,
+        String detail,
+        String instance,
+        Object properties
+    ) {}
+
+    @Test
+    @DisplayName("Create a new todo with extra field and without description - success request")
+    void createTodo_ExtraFieldAndEmptyDescription_Success() throws Exception {
+        // Arrange
+        String invalidRequest = """
```

Comment:
That great that you are using text blocks! Also, as an option, we can store request data in files and read them from the resource folders (this is optional, but you can use this approach for a big request DTO objects in future)